gips/0082-substreams-on-the-network.md

```diff
@@ -0,0 +1,106 @@
+Given that there's no gateway involved with Substreams (it's direct point-to-point between consumer and provider), the existing Indexer Selection Algorithm (ISA) does not apply.  A round-robin selection of providers will be used in the front-end (https://thegraph.market) to ensure fair distribution of request loads, while still allowing consumers to choose a provider.
+
+## Payments
+
+Payments for Substreams work will be on-chain. The existing `collect` function in the staking contract will be used for payment processing.  It was deemed that no TAP was required to have Substreams collect payments and honor the promises of the protocol. However, as TAP brings augmented trust minimization properties, it would be incorporated in future work.
+
+## Economic Security and Disputes
+
+Economic security will be enforced through slashing and disputes. Indexers providing incorrect data can have their stake slashed.
+
+- **Attestations:** Substreams endpoints will provide signed attestations to verify data integrity. The command-line tools (`substreams gui`) will display these attestations.  To accommodate different future attestation methods, and not only an Ethereum key signature, the attestation payload will be prefixed with the protocol used (e.g., `eth:` for Ethereum). Detailed documentation will describe the attestation format and verification process.
+- **Detecting Faults:** Since Substreams are streaming, Proofs of Indexing (PoIs) are not applicable. Fault detection will happen at query time. Discrepancies in data returned by different providers for the same stream and block will trigger an investigation by an Arbiter. The Arbiter will use the `substreams gui` tool to inspect and compare stream data at specific block ranges, verifying data integrity and determinism.
+
+In the `SessionInit` message of the response, when requiring SignedAttestation, the server would reply with the `attestation_address string` protocol would be `eth`, and for future proofing, and `address` would be an `eth` address (indexer address or operator address, whatever is less risky)
+
+### Operator keys for signing
+
+Registration will be done by the staking address. The operator key will be used for signing, and an in-memory mapping will link the operator key to the staking key.
+- The `setOperator` function ([Staking.sol#L226](https://github.com/graphprotocol/contracts/blob/ce3ec16484dacc89a1b9cf08455256be830790e3/packages/contracts/contracts/staking/Staking.sol#L226)) links the operator key to the staking key and emits the `SetOperator` event ([IStakingBase.sol#L99](https://github.com/graphprotocol/contracts/blob/main/packages/contracts/contracts/staking/IStakingBase.sol#L99)).
+
+
+
+## Arbitration
+
+StreamingFast will provide support for arbitration cases involving Substreams, either by hiring an Arbiter or by collaborating with individuals experienced in arbitration.  A detailed document outlining the arbitration process for Substreams discrepancies will be created.
```

Comment:
I think some detail about how arbitration is going to work is needed here. It sounds like its going to be a completely off chain process, however the slashing part needs to occur on-chain through the staking contract. How is that going to be bridged?